### PR TITLE
Fix a bug for gpt pre-training. 

### PR DESCRIPTION
**Pull request by FeixLiu**
For GPT pre-training with `examples/pretrain_gpt_distributed.sh` commands. It will encounter an error like this.
c727e6e174c6794195549064b27e9046

It seems that the tuple `outputs` here (https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/mpu/random.py#L311) contains a redundant torch tensor at the end of the tuple, which will cause the error above.
image

By removing the last value of the `outputs`, the pre-train can be run through. 
image

**This PR is a straight forward solution for the problem. If there is a better way, please let me know.**

### CODE REVIEW

**Comment by conglongli** [comment]
The extra tensor is probably because we added an auxiliary loss to support MoE model. On the other hand, you hit this error because you used a very old script from original Megatron repo that doesn't have DeepSpeed integration (with DeepSpeed support, you won't touch this part of code), which is the case we don't guarantee to work since this repo mainly focus on showcase of DeepSpeed features (see https://github.com/microsoft/Megatron-DeepSpeed/blob/main/examples/README.md). If you want to pretrain gpt model I recommend starting from https://github.com/microsoft/Megatron-DeepSpeed/blob/main/examples/MoE/ds_pretrain_gpt_350M_dense.sh (make sure to install DeepSpeed). Regarding this PR, since it is a quite hacky solution, please use an elif to limit it to the condition to only when there are 2 tensors and only when the second tensor has a single zero value, and please add a comment that this is a hacky solution to overcome issue when running a very old script examples/pretrain_gpt_distributed.sh that doesn't have proper DeepSpeed integration.

**Comment by FeixLiu** [comment]
> conglongli

Thanks for the reply. Thanks for the reply. Have updated the codes follow the comment.

**Review by conglongli — APPROVED** [review]
